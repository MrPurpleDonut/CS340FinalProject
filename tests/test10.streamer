text data << args[0];
list<text> lines = data / "\n";
list<text> matches = lines[? args[1] ?];
print matches;
