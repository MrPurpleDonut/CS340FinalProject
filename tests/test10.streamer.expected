Program parsed correctly!
---------------------------
--------Printing AST-------
---------------------------
Program
    Function List
    Statement List
        Declaration statement: data
            Type: text
            File Read into variable: data
                List Access
                    Identifier: args
                    Number: 0.0
        Declaration statement: lines
            Type: list
                Type: text
            Assignment statement: lines
                Binary operator '/'
                    Identifier: data
                    Quoted string: "\n"
        Declaration statement: matches
            Type: list
                Type: text
            Assignment statement: matches
                List Filter
                    Identifier: lines
                    List Access
                        Identifier: args
                        Number: 1.0
        Print Statement
            Identifier: matches
